### ISSUE TITLE
macOS case-sensitive "failed to satisfy dependency ddApp"

### ISSUE THREAD

**Issue by goretkin** [rdeits/DrakeVisualizer.jl#58]
I came across this error when trying to build [RigidBodyTreeInspector](https://github.com/rdeits/RigidBodyTreeInspector.jl)

```
INFO: Attempting to Create directory /Users/goretkin/.julia/v0.6/DrakeVisualizer/deps/usr
==============================================================================[ ERROR: DrakeVisualizer ]==============================================================================

LoadError: Provider BinDeps.BuildProcess failed to satisfy dependency ddApp
while loading /Users/goretkin/.julia/v0.6/DrakeVisualizer/deps/build.jl, in expression starting on line 95

============================================================================================================
```

It looks like the problem is because I'm on macOS, with a case-sensitive filesystem

```
goretkin$ pwd
/Users/goretkin/.julia/v0.6/DrakeVisualizer/deps/usr
goretkin$ ls -la
total 32
drwxr-xr-x  3 goretkin  staff   238B Jul  6 12:46 ./
drwxr-xr-x  4 goretkin  staff   204B Jul  6 12:46 ../
drwxr-xr-x  3 goretkin  staff   102B Mar  1 20:30 Director.app/
lrwxr-xr-x  1 goretkin  staff    31B Mar  1 20:33 bin@ -> Director.app/Contents/MacOs/bin
lrwxr-xr-x  1 goretkin  staff    35B Mar  1 20:33 include@ -> Director.app/Contents/MacOs/include
lrwxr-xr-x  1 goretkin  staff    31B Mar  1 20:33 lib@ -> Director.app/Contents/MacOs/lib
lrwxr-xr-x  1 goretkin  staff    33B Mar  1 20:33 share@ -> Director.app/Contents/MacOs/share
```

those `MacOs`s should be `MacOS`.

Not sure which piece of code is responsible for making that directory, but manually fixing the symlinks and running `Pkg.build("DrakeVisualizer")` seemed to work for me.

**Comment by rdeits** [comment]
Oo, macOS with case-sensitive files? That's...adventurous 😉 

Could you try repeating this process with DrakeVisualizer master? That is, do:

```
using DrakeVisualizer
DrakeVisualizer.delete_director_binaries()
Pkg.checkout("DrakeVisualizer")
Pkg.build("DrakeVisualizer")
```

the Director build has changed since the version you're using, so I'd like to see if this is still an issue.

**Comment by goretkin** [comment]
Yeah, if I get bitten enough, maybe I'll change my mind ;)

For some reason, I can't run `using DrakeVisualizer` (I was using the package before through `RigidBodyTreeInspector`). 

```
julia> using DrakeVisualizer
INFO: Precompiling module DrakeVisualizer.
INFO: Recompiling stale cache file /Users/goretkin/.julia/lib/v0.6/GeometryTypes.ji for module GeometryTypes.
WARNING: The call to compilecache failed to create a usable precompiled cache file for module GeometryTypes. Got:
WARNING: Module Iterators uuid did not match cache file.
ERROR: LoadError: Declaring __precompile__(true) is only allowed in module files being imported.
Stacktrace:
 [1] __precompile__(::Bool) at ./loading.jl:335
 [2] __precompile__() at ./loading.jl:331
 [3] include_from_node1(::String) at ./loading.jl:569
 [4] eval(::Module, ::Any) at ./boot.jl:235
 [5] _require(::Symbol) at ./loading.jl:483
 [6] require(::Symbol) at ./loading.jl:398
 [7] _include_from_serialized(::String) at ./loading.jl:157
 [8] _require_from_serialized(::Int64, ::Symbol, ::String, ::Bool) at ./loading.jl:200
 [9] _require(::Symbol) at ./loading.jl:491
 [10] require(::Symbol) at ./loading.jl:398
while loading /Users/goretkin/.julia/v0.6/GeometryTypes/src/GeometryTypes.jl, in expression starting on line 1
```

In any case, I deleted the files manually. 

The director version downloaded is `http://people.csail.mit.edu/patmarion/software/director/releases/director-0.1.0-130-g4109097-mac.tar.gz` (previously it was `director-0.1.0-35-g26aa674-mac.tar.gz`)

The error is now
```
INFO: Attempting to Create directory /Users/goretkin/.julia/v0.6/DrakeVisualizer/deps/usr
==============================================================================[ ERROR: DrakeVisualizer ]==============================================================================

LoadError: Provider BinDeps.BuildProcess failed to satisfy dependency vtkDRCFilters
while loading /Users/goretkin/.julia/v0.6/DrakeVisualizer/deps/build.jl, in expression starting on line 97

======================================================================================================================================================================================
```

and the issue appears to be the same

```
~ goretkin$ ls -la /Users/goretkin/.julia/v0.6/DrakeVisualizer/deps/usr
total 32
drwxr-xr-x  3 goretkin  staff   238B Jul  6 14:21 ./
drwxr-xr-x  4 goretkin  staff   204B Jul  6 14:21 ../
drwxr-xr-x  3 goretkin  staff   102B Jun 13 16:15 Director.app/
lrwxr-xr-x  1 goretkin  staff    31B Jun 13 16:21 bin@ -> Director.app/Contents/MacOs/bin
lrwxr-xr-x  1 goretkin  staff    35B Jun 13 16:21 include@ -> Director.app/Contents/MacOs/include
lrwxr-xr-x  1 goretkin  staff    31B Jun 13 16:21 lib@ -> Director.app/Contents/MacOs/lib
lrwxr-xr-x  1 goretkin  staff    33B Jun 13 16:21 share@ -> Director.app/Contents/MacOs/share
```

**Comment by rdeits** [comment]
Hm, OK. This is probably actually an issue with the Director packaging script, so let's move it over there: https://github.com/RobotLocomotion/director/issues/532

**Comment by goretkin** [comment]
I tried to verify that the fix in https://github.com/RobotLocomotion/director/commit/7ce9a5aa02df24e320c63639b4cc36c77bfe6b65 would work by building from source (and updating the `director_sha` in `build.jl`. However it appears the build script for `is_apple()` doesn't support building from source.

I tried to modify it, not quite knowing what I'm doing, https://github.com/goretkin/DrakeVisualizer.jl/commit/57524e8fbfb317802c17e25781bb7b773f4c7142 and brew quietly ran for a _long time_ before 

```
==============================================================================[ ERROR: DrakeVisualizer ]==============================================================================

LoadError: failed process: Process(`/Users/goretkin/.julia/v0.6/Homebrew/deps/usr/bin/brew install cmake python numpy qt vtk7 eigen`, ProcessExited(1)) [1]
while loading /Users/goretkin/.julia/v0.6/DrakeVisualizer/deps/build.jl, in expression starting on line 104
```

Before that there were a few warnings
```
Warning: homebrew/python was deprecated. This tap is now empty as all its formulae were migrated.
Cloning into '/Users/goretkin/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-python'...
remote: Counting objects: 5, done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 5 (delta 0), reused 4 (delta 0), pack-reused 0
Unpacking objects: 100% (5/5), done.
Cloning into '/Users/goretkin/.julia/v0.6/Homebrew/deps/usr/Library/Taps/patmarion/homebrew-director'...
remote: Counting objects: 7, done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 7 (delta 0), reused 4 (delta 0), pack-reused 0
Unpacking objects: 100% (7/7), done.
Warning: qt is provided by core, but is now shadowed by patmarion/director/qt.
To refer to the core formula, use Homebrew/core/qt instead.
```

**Comment by rdeits** [comment]
Hm, yeah, looks like those instructions might be out of date. I'll check with pat: https://github.com/RobotLocomotion/director/issues/534

**Comment by rdeits** [comment]
Also, new binaries are already available in #59 if you want to try that
